common/network_manager.go

```diff
@@ -419,14 +430,22 @@ func (m *NetworkManager) failOrContinueRequest(event *fetch.EventRequestPaused,
 	}
 }
 
-func handleBlockedHosts(u *url.URL, blockedHosts *k6types.HostnameTrie) string {
-	ip := net.ParseIP(u.Host)
-	if ip == nil {
-		if match, blocked := blockedHosts.Contains(u.Host); blocked {
-			return fmt.Sprintf("hostname %s is in a blocked pattern (%s)", u.Host, match)
+func checkBlockedHosts(host string, blockedHosts *k6types.HostnameTrie) error {
+	if match, blocked := blockedHosts.Contains(host); blocked {
+		return fmt.Errorf("hostname %s is in a blocked pattern (%s)", host, match)
+	}
+	return nil
+}
+
+func checkBlockedIPs(ip net.IP, blockedIPs []*k6lib.IPNet) error {
+	for _, ipnet := range blockedIPs {
+		if ipnet.Contains(ip) {
```

Comment:
Note that this doesn't do a lookup to block the IP a host resolves to (as [this comment implied](https://github.com/grafana/xk6-browser/commit/47b0121ff97e9fb2d8b5e004f82c785044598924/#diff-30b70a5056bb318589cf41ae368bc837cc814b2f2d7bd0fbd24cb7fc2ed0294fL381)), just does a basic match.

I don't see a way we can get the resolved IP at this stage, so we might have to do an additional lookup if we want this feature. Should we if it's not possible otherwise?

Reply:
True, and I think we should do a lookup, yes. Shouldn't add much overhead time-wise. Would perhaps interfere with DNS lookup timings (when the actual requests happens in the browser) if the lookup is cached locally as a result of this blocklist check?

Reply:
It could definitely mess with the timings, but the caches would likely be different, since here we'll use Go's built-in resolver (which could also be forced to use `getaddrinfo` via cgo), and the browser has its own internal cache. And I'm not sure what's the status of supporting e.g. `/etc/resolv.conf`, etc.

Since this is an interoperability mess, I'm slightly concerned we might run into a situation where our lookup returns one result, and the browser's returns another. :man_shrugging: I guess we'll cross that bridge when we get there, but it's not out of the question we might have to.

I'll see if there's another way of getting the IP, otherwise I'll add the lookup here.

Reply:
This should be _resolved by_ ;) 0cc6e6b. We have to create our own `Resolver` instance since the k6 one is not accessible to extensions :disappointed:, but it should be identical in behavior as the k6 one, since it's built with the same DNS options.

@robingustafsson Let me know if there's anything pending here.